### PR TITLE
[stable/redis] add RBAC

### PR DESCRIPTION
**Pull request by barryib**
**What this PR does / why we need it**:
Add RBAC (serviceAccount, Role, Rolebinding) option to the sbale/redis chart.

/cc @prydonius @tompizmor @sameersbn @carrodher @juan131

### CODE REVIEW

**Comment by prydonius** [comment]
@barryib AFAIK Redis does not use the k8s API so I can't see why this change is necessary, anything I'm missing?

**Comment by barryib** [comment]
@prydonius In our k8s clusters, every pod should run with a service account and should validate against pod security policy. I got error like this:

```
Events:
  Type     Reason        Age                From                    Message
  ----     ------        ----               ----                    -------
  Warning  FailedCreate  5s (x12 over 16s)  statefulset-controller  create Pod redis-master-0 in StatefulSet redis-master failed error: pods "redis-master-0" is forbidden: unable to validate against any pod security policy: []
```
So with this PR, redis chart will create a service account and bind it to the  role with the given rules:

```yaml
redis:
  serviceAccount:
    create: true
  rbac:
    create: true
    role:
      rules:
        - apiGroups:
            - extensions
          resources:
            - podsecuritypolicies
          verbs:
            - use
          resourceNames:
            - bme.unprivileged
```

File: stable/redis/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: redis
-version: 3.3.5
+version: 3.3.6
```
Comment (prydonius) [inline stable/redis/Chart.yaml]: please bump this to 3.4.0

**Review by prydonius — COMMENTED** [review]
interesting, thanks for the information @barryib. @unguiculus what do you think about this? It sounds like we may need to add this to all charts if we add it here.

File: stable/redis/Chart.yaml (hunk 1 above)
Comment (barryib) [reply stable/redis/Chart.yaml]: @prydonius done with 9edcea6

File: stable/redis/templates/redis-master-statefulset.yaml [hunk 2]
```diff
@@ -39,6 +39,9 @@ spec:
         fsGroup: {{ .Values.master.securityContext.fsGroup }}
         runAsUser: {{ .Values.master.securityContext.runAsUser }}
       {{- end }}
+      {{- if .Values.serviceAccount.create }}
```
Comment (unguiculus) [inline stable/redis/templates/redis-master-statefulset.yaml]: No need to wrap into an if block.

File: stable/redis/templates/redis-slave-deployment.yaml [hunk 3]
```diff
@@ -34,6 +34,9 @@ spec:
       {{- end}}
       {{- /* Include master securityContext if slave securityContext not defined */ -}}
       {{ include "redis.slave.securityContext" . | indent 6 }}
+      {{- if .Values.serviceAccount.create }}
```
Comment (unguiculus) [inline stable/redis/templates/redis-slave-deployment.yaml]: No need to wrap into an if block.

**Review by unguiculus — CHANGES_REQUESTED** [review]
A couple of comments. Otherwise LGTM.

File: stable/redis/values.yaml [hunk 4]
```diff
@@ -64,6 +64,30 @@ networkPolicy:
...
+serviceAccount:
+  # Specifies whether a ServiceAccount should be created
+  create: false
+  # The name of the ServiceAccount to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name:
+
+rbac:
+  # Specifies whether RBAC resources should be created
+  create: false
```
Comment (juan131) [inline stable/redis/values.yaml:76]: I would use `enabled` as we do for other components such as **livenessProbes**, **readinessProbes**, etc.

File: stable/redis/values.yaml [hunk 5]
```diff
@@ -64,6 +64,30 @@ networkPolicy:
   ##
   # allowExternal: true
 
+serviceAccount:
+  # Specifies whether a ServiceAccount should be created
+  create: false
```
Comment (juan131) [inline stable/redis/values.yaml:69]: I would use `enabled` as we do for other components such as **livenessProbes**, readinessProbes, etc.

**Review by juan131 — COMMENTED** [review]
Just a couple of comments about the wording, but it's ok to me

File: stable/redis/values.yaml (hunk 4 above)
Comment (juan131) [reply stable/redis/values.yaml:76]: Forget about this comment. As it's documented at https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/conventions.md, `create` is fine.

File: stable/redis/values.yaml (hunk 5 above)
Comment (juan131) [reply stable/redis/values.yaml:69]: Forget about this comment. As it's documented at https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/conventions.md, `create` is fine.

File: stable/redis/templates/redis-slave-deployment.yaml (hunk 3 above)
Comment (barryib) [reply stable/redis/templates/redis-slave-deployment.yaml]: @unguiculus removing the if block will unconditionally ’serviceAcvount’ even if the serviceAccountName is empty. Furthermore, it'll pod will try to run with a possibly inexistent service account.

File: stable/redis/templates/redis-master-statefulset.yaml (hunk 2 above)
Comment (barryib) [reply stable/redis/templates/redis-master-statefulset.yaml]: See comment below.

Comment (barryib) [reply stable/redis/templates/redis-master-statefulset.yaml]: @unguiculus Forget about precedent comment. Fixed in cb669279938dc9f22f48118e0ff239a83d6b6e24

File: stable/redis/templates/redis-slave-deployment.yaml (hunk 3 above)
Comment (barryib) [reply stable/redis/templates/redis-slave-deployment.yaml]: @unguiculus Forget about precedent comment. Fixed in cb669279938dc9f22f48118e0ff239a83d6b6e24

**Pull request merged**